primitives/core/src/offchain/mod.rs

```diff
@@ -195,6 +195,7 @@ pub struct OpaqueNetworkState {
 
 /// Simple blob to hold a `Multiaddr` without committing to its format.
 #[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug, PassByInner, TypeInfo)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
 pub struct OpaqueMultiaddr(pub Vec<u8>);
```

Comment:
```suggestion
pub struct OpaqueMultiaddr(pub crate::Bytes);
```

We should do this when you want to derive serde.

Reply:
Hi, `crate::Bytes` cause the problem